### ISSUE TITLE
java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter

### ISSUE THREAD

**Issue by nagylzs** [Yubico/yubico-java-client#21]
The latest version 3.0.4 depends on javax.xml.bind which is not present in JRE 11.

Actually, it was already deprecated in JRE 9:

https://docs.oracle.com/javase/9/docs/api/java.xml.bind-summary.html

and it was completely removed in JRE 11:

https://docs.oracle.com/en/java/javase/11/migrate/index.html#JSMIG-GUID-561005C1-12BB-455C-AD41-00455CAD23A6

Unfortunately, this will not result in a compile time error, because the java.xml.bind package was never listed as a dependency in pom.xml (being a system package). But it results in a runtime error, if you try to use the validation client.

```

 java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1364)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1185)
	at com.yubico.client.v2.YubicoClient.setKey(YubicoClient.java:102)
	at com.yubico.client.v2.impl.YubicoClientImpl.<init>(YubicoClientImpl.java:82)
	at com.yubico.client.v2.YubicoClient.getClient(YubicoClient.java:156)

```

**Comment by emlun** [comment]
Thanks for the report! This looks easy enough to solve for JRE 8+, because we're only using `DataTypeConverter` for Base64 codecing, but we'll have to think a bit about what to do about JRE 7.

**Comment by gir489** [comment]
@emlun I feel that since I personally made this change, I should chime in with a solution. We could do one of the following:

1. Reimplement DatatypeConverter manually within the project using an internal class.
2. Go back to the old Apache dependency.
3. Use compile time switches for when compiling against JDK9+ and using the newer libraries.

**Comment by emlun** [comment]
Oh right, that was in #13. I'd forgotten that we used to have commons-codec as a dependency earlier. In that case I'll just revert to using commons-codec for now, and then we can drop it again after JRE 7 reaches end of life.

**Comment by emlun** [comment]
Fixed in 57a88577dd0be3158e9f9c29310cf74c7c4fd163; v3.0.5 entering release pipeline now.
